packages/accounts/src/light-account/utils.ts

```diff
@@ -15,114 +14,294 @@ import {
+            "0x19B486A57C790b47Ee2E967c2aB1b8BbAeAeE7bB".toLowerCase() as Address,
+          impl: "0x118c4EA651e2fE5F5637f5D8A473CB8329251f89".toLowerCase() as Address,
+        })
+      ),
+      entryPointVersion: "0.7.0",
+    },
   },
-  "v1.1.0": {
-    factoryAddress:
-      "0x00004EC70002a32400f8ae005A26081065620D20".toLowerCase() as Address,
-    implAddress:
-      "0xae8c656ad28F2B59a196AB61815C16A0AE1c3cba".toLowerCase() as Address,
+  MultiOwnerLightAccount: {
+    "v2.0.0": {
+      type: "MultiOwnerLightAccount",
+      version: "v2.0.0",
+      address: toRecord<Chain, "id", { factory: Address; impl: Address }>(
+        [arbitrumSepolia],
+        "id",
+        () => ({
+          factory:
+            "0xEEeB9dD9DA59bD3020C548D3543485E37ee8A50e".toLowerCase() as Address,
+          impl: "0x0C93750D969bdeb51547766680E5CA14A508758e".toLowerCase() as Address,
+        })
+      ),
+      entryPointVersion: "0.7.0",
+    },
   },
 };
 
+export const defaultLightAccountVersion = <
+  TLightAccountType extends LightAccountType
+>(
+  type: TLightAccountType
+): LightAccountVersion<TLightAccountType> =>
+  (type === "LightAccount"
+    ? "v1.1.0"
+    : "v2.0.0") as LightAccountVersion<TLightAccountType>;
+
 /**
- * Utility method returning the default light account factory address given a {@link Chain} object
+ * Example Usage:
  *
- * @param chain - a {@link Chain} object
- * @returns a {@link Address} for the given chain
- * @throws if the chain doesn't have an address currently deployed
+ * type T1 = GetLightAccountVersionsForType<"LightAccount">;
+ * const t1: T1 = AccountVersionRegistry.LightAccount["v1.0.2"];
  */
-export const getDefaultLightAccountFactoryAddress = (
-  chain: Chain,
-  version: LightAccountVersion = "v1.1.0"
-): Address => {
-  switch (chain.id) {
-    case mainnet.id:
-    case sepolia.id:
-    case goerli.id:
-    case polygon.id:
-    case polygonAmoy.id:
-    case polygonMumbai.id:
-    case optimism.id:
-    case optimismGoerli.id:
-    case optimismSepolia.id:
-    case arbitrum.id:
-    case arbitrumGoerli.id:
-    case arbitrumSepolia.id:
-    case base.id:
-    case baseGoerli.id:
-    case baseSepolia.id:
-      return LightAccountVersions[version].factoryAddress;
-  }
+export type GetLightAccountVersionDefsForType<TType extends LightAccountType> =
+  Extract<
+    IAccountVersionRegistry[TType][keyof IAccountVersionRegistry[TType]],
+    { type: TType }
+  >;
 
-  throw new DefaultFactoryNotDefinedError("LightAccount", chain, "0.6.0");
-};
+export type GetLightAccountVersionDef<
+  TType extends LightAccountType,
+  TLightAccountVersion extends LightAccountVersion<TType>
+> = IAccountVersionRegistry[TType][TLightAccountVersion];
+
+/**
+ * Example Usage:
+ *
+ * type T = GetLightAccountVersionDefsForEntryPoint<"LightAccount", "0.6.0">;
+ * const t_1: T = AccountVersionRegistry.LightAccount["v1.0.2"]; // compiles
+ * const t_2: T = AccountVersionRegistry.LightAccount["v2.0.0"]; // errors
```

Comment:
jsdocs supports passing in examples so that they're well formatted

you can use the `@example` syntax

Reply:
https://jsdoc.app/tags-example

Reply:
updated.